sqlx-postgres/src/types/int.rs

```diff
@@ -6,6 +6,16 @@ use crate::error::BoxDynError;
 use crate::types::Type;
 use crate::{PgArgumentBuffer, PgHasArrayType, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
 
+fn int_decode(value: PgValueRef<'_>) -> Result<i64, BoxDynError> {
+    Ok(match value.format() {
+        PgValueFormat::Text => value.as_str()?.parse()?,
+        PgValueFormat::Binary => {
+            let buf = value.as_bytes()?;
+            BigEndian::read_int(buf, buf.len())
```

Comment:
This will panic if `buf` is empty or is more than 8 bytes (same bug as #2691). It should check that and return an error instead.

Reply:
It shouldn't happen in normal operation but sometimes people try to decode the wrong type.